sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client_helpers.py

```diff
@@ -0,0 +1,1351 @@
+
+    validate_content = kwargs.pop('validate_content', False)
+    content_settings = kwargs.pop('content_settings', None)
+    overwrite = kwargs.pop('overwrite', False)
+    max_concurrency = kwargs.pop('max_concurrency', 1)
+    cpk = kwargs.pop('cpk', None)
+    cpk_info = None
+    if cpk:
+        if scheme.lower() != 'https':
+            raise ValueError("Customer provided encryption key must be used over HTTPS.")
+        cpk_info = CpkInfo(encryption_key=cpk.key_value, encryption_key_sha256=cpk.key_hash,
+                            encryption_algorithm=cpk.algorithm)
+    kwargs['cpk_info'] = cpk_info
+
+    headers = kwargs.pop('headers', {})
+    headers.update(add_metadata_headers(metadata))
+    kwargs['lease_access_conditions'] = get_access_conditions(kwargs.pop('lease', None))
+    kwargs['modified_access_conditions'] = get_modify_conditions(kwargs)
+    kwargs['cpk_scope_info'] = get_cpk_scope_info(kwargs)
+    if content_settings:
+        kwargs['blob_headers'] = BlobHTTPHeaders(
+            blob_cache_control=content_settings.cache_control,
+            blob_content_type=content_settings.content_type,
+            blob_content_md5=content_settings.content_md5,
+            blob_content_encoding=content_settings.content_encoding,
+            blob_content_language=content_settings.content_language,
+            blob_content_disposition=content_settings.content_disposition
+        )
+    kwargs['blob_tags_string'] = serialize_blob_tags_header(kwargs.pop('tags', None))
+    kwargs['stream'] = stream
+    kwargs['length'] = length
+    kwargs['overwrite'] = overwrite
+    kwargs['headers'] = headers
+    kwargs['validate_content'] = validate_content
+    kwargs['blob_settings'] = config
+    kwargs['max_concurrency'] = max_concurrency
+    kwargs['encryption_options'] = encryption_options
+    # Add feature flag to user agent for encryption
+    if key_encryption_key:
+        modify_user_agent_for_encryption(
+            config.user_agent_policy.user_agent,
+            sdk_moniker,
+            encryption_version,
+            kwargs)
+
+    if blob_type == BlobType.BlockBlob:
+        kwargs['client'] = client.block_blob
+        kwargs['data'] = data
+    elif blob_type == BlobType.PageBlob:
+        if encryption_version == '2.0' and (require_encryption or key_encryption_key is not None):
+            raise ValueError("Encryption version 2.0 does not currently support page blobs.")
+        kwargs['client'] = client.page_blob
+    elif blob_type == BlobType.AppendBlob:
+        if require_encryption or (key_encryption_key is not None):
+            raise ValueError(_ERROR_UNSUPPORTED_METHOD_FOR_ENCRYPTION)
+        kwargs['client'] = client.append_blob
+    else:
+        raise ValueError(f"Unsupported BlobType: {blob_type}")
+    return kwargs
+
+def _upload_blob_from_url_options(
+    scheme,
+    source_url,
+    **kwargs):
+    # type: (...) -> Dict[str, Any]
```

Comment:
So I see you aren't really adding types to this file right now, and that's probably fine but maybe we should at least not introduce incorrect type statements? Like this existing `# type: (...) -> Dict[str, Any]` is now just wrong since you added parameters.

This is going into a feature branch so it probably doesn't matter but would make me feel better if you just at least deleted this old type hints if you don't want to fill in the types now (which I think is fine) :)